### PR TITLE
Upgrade to vue3 

### PR DESCRIPTION
**Pull request by xiaofan9**
vue底层从2.x升级为vue3。

### CODE REVIEW

File: Gulpfile.js [hunk 1]
```diff
@@ -1,16 +1,28 @@
-const gulp = require('gulp');
+
```
Comment (imcvampire) [inline Gulpfile.js:1]:
```suggestion
```

Please remove this empty line to keep everything clear

File: src/index.js [hunk 2]
```diff
@@ -45,5 +31,4 @@ if (typeof exports == "object") {
 } else if (window.Vue && window.axios) {
   Vue.use(plugin, window.axios)
```
Comment (imcvampire) [inline src/index.js:32]: Does this line still know with Vue 3?

File: index.d.ts [hunk 3]
```diff
@@ -1,20 +1,18 @@
-import Vue, {PluginFunction, PluginObject} from "vue";
 import { AxiosStatic } from "axios";
+import { App } from "vue";
 
-declare module "vue/types/vue" {
-
-  interface Vue {
-    axios: AxiosStatic;
+export declare function VAxios(app: App): AxiosStatic;
```
Comment (imcvampire) [inline index.d.ts:4]:
```suggestion
export declare function VueAxios(app: App): AxiosStatic;
```

Comment (imcvampire) [inline index.d.ts:4]: There is no implementation for this function?

File: src/index.js (hunk 2 above)
Comment (xiaofan9) [reply src/index.js:32]: Yes, I have tested it, and there is no problem with import

File: index.d.ts (hunk 3 above)
Comment (xiaofan9) [reply index.d.ts:4]: yes,this is my mistake

**Comment by imcvampire** [comment]
Is this MR finished? @xiaofan9

**Comment by xiaofan9** [comment]
> Is this MR finished? @xiaofan9
> 
> It's still draft now

this MR finished，but when I try pr, it doesn’t seem to pass！

**Comment by xiaofan9** [comment]
This MR has been completed. I have tested it and it is no problem @imcvampire

File: Gulpfile.js [hunk 4]
```diff
@@ -1,16 +1,27 @@
...
+const { src, dest, series } = require('gulp');
 const babel = require('gulp-babel');
 const uglify = require('gulp-uglifyjs');
 const rename = require('gulp-rename');
+const clean = require('gulp-clean');
 
-gulp.task('default', () => {
-  return gulp.src('src/index.js')
+function build() {
+  return src('src/index.js', { allowEmpty: true })
```
Comment (imcvampire) [inline Gulpfile.js:8]:
```suggestion
  return src('src/index.js')
```

`allowEmpty` doesn't seem a good option here. We should catch some human errors, one of them maybe is an empty file

File: src/index.js [hunk 5]
```diff
@@ -2,40 +2,26 @@
 
 /**
  * Install plugin
- * @param Vue
+ * @param app
```
Comment (imcvampire) [inline src/index.js:5]: Is there any way to mention this param is a Vue instance?

File: Gulpfile.js (hunk 4 above)
Comment (xiaofan9) [reply Gulpfile.js:8]: ok

File: src/index.js (hunk 5 above)
Comment (xiaofan9) [reply src/index.js:5]: It was my oversight

**Review by imcvampire — APPROVED** [review]

**Pull request merged**